pycc/lccwfn.py

```diff
@@ -0,0 +1,476 @@
+import time 
+#from timer import Timer
+import numpy as np
+from opt_einsum import contract  
+
+
+class lccwfn(object):
+    """
+    A PNO-CC object. 
+
+    Attributes
+    ----------
+    o : Numpy slice
+        occupied orbital subspace 
+    v : Numpy slice
+        virtual orbital subspace
+    no: int
+        number of (active) occupied orbitals
+    nv: int 
+        number of virtual orbitals
+    H: PyCC Hamiltonian object
+       Hamiltonian object with integrals, Fock matrices, and orbital coefficients
+    Local: PyCC Local object 
+       Local object with transformation matrices, local Fock matrices and integrals, etc. 
+      
+    Parameters
+    ----------
+    local: string 
+           type of local calculation ("PAO", "PNO", etc.)
+    model: string
+           type of CC calculation ("CCD","CCSD", etc.)
+    eref: float
+          reference energy (typically HF/SCF energy)     
+
+    Methods
+    -------
+    solve_lcc() 
+        Solves the local CC T amplitude equations
+    local_residuals()
+        Computes the local T1 and T2 residuals for a given set of amplitudes and Fock operator  
+    
+    Notes
+    -----
+    To do: 
+    (1) need DIIS extrapolation
+    (2) time table for each intermediate?
+    (3) generate and store overlap terms prior to the calculation of the residuals
+    (4) remove redundant transformed integrals 
+    """
+ 
+    def __init__(self, o, v, no, nv, H, local, model, eref, Local): 
+        self.o = o
+        self.v = v
+        self.no = no
+        self.nv = nv
+        self.H = H
+        self.local = local
+        self.model = model 
+        self.eref = eref
+        self.Local = Local 
+        self.QL = self.Local.QL
+        self.dim = self.Local.dim
+        self.eps = self.Local.eps 
+
+        self.t1 = np.zeros((self.no, self.nv))
+        t1_ii = []
+        t2_ij = [] 
+
+        for i in range(self.no):
+            ii = i*self.no + i
+    
+            t1_ii.append(self.QL[ii].T @ self.t1[i]) 
```

Comment:
Is this just creating a vector of zeroes the length of the PNO space for pair ii?  If so, perhaps it would be better to generate this in a simpler manner?

Reply:
Yes, it's just that. I can simplify it further by coding up instead in the `i loop` something like `t1.append(np.zeros((self.Local.dim[ii])))`